src/main/java/com/fasterxml/jackson/databind/deser/std/PrimitiveArrayDeserializers.java

```diff
@@ -214,7 +214,7 @@ public T deserialize(JsonParser p, DeserializationContext ctxt, T existing) thro
     protected T handleNonArray(JsonParser p, DeserializationContext ctxt) throws IOException
     {
         // Empty String can become null...
-        if (p.hasToken(JsonToken.VALUE_STRING)) {
+        if (p.hasToken(JsonToken.VALUE_STRING) && p.getText().isEmpty()) {
             return _deserializeFromString(p, ctxt);
```

Comment:
@eeren-bm Have a look at this method in `StdDeserializer`... By-passing it will skip lots of potentially necessary processing, including `CoercionConfig` settings.

But I think we can get this to work by bit more re-factoring. Let me see if I can change it a bit.